libs/vkd3d/breadcrumbs.c

```diff
@@ -468,6 +600,16 @@ void vkd3d_breadcrumb_tracer_signal(struct d3d12_command_list *list)
                         offsetof(struct vkd3d_breadcrumb_counter, begin_marker),
                 trace->counter + 1));
     }
+    else
+    {
+        checkpoint = vkd3d_malloc(sizeof(struct vkd3d_nv_checkpoint));
+        checkpoint->context_index = context;
+        checkpoint->counter = trace->counter;
+        VK_CALL(vkCmdSetCheckpointNV(list->vk_command_buffer, checkpoint));
```

Comment:
It seems very pessimistic and slow to allocate a tiny cookie for each check point. Ideally, just encode the context index + breadcrumb index as a pointer directly. Even 32-bit pointer should be enough.